test/easyconfigs/easyconfigs.py

```diff
@@ -1315,6 +1315,9 @@ def template_easyconfig_test(self, spec):
     # sanity check for software name, moduleclass
     self.assertEqual(ec['name'], name)
     self.assertTrue(ec['moduleclass'] in build_option('valid_module_classes'))
+    # base is the default value for moduleclass, which should never be used,
+    # and moduleclass should always be set in the easyconfig file
+    self.assertNotEqual(ec['moduleclass'], 'base', "moduleclass should be set, and not be set to 'base', for %s", spec)
```

Comment:
```suggestion
    self.assertNotEqual(ec['moduleclass'], 'base',
    "moduleclass should be set, and not be set to 'base', for %s" % spec)
```

Hopefully I've got the correct code this time